subprojects/language-native/src/main/java/org/gradle/language/swift/plugins/SwiftBasePlugin.java

```diff
@@ -116,6 +118,22 @@ public String call() {
                     link.setOutputFile(runtimeFile);
                     link.setTargetPlatform(currentPlatform);
                     link.setToolChain(toolChain);
+                } else if (binary instanceof SwiftBundle) {
+                    // Add a link task
+                    LinkBundle link = tasks.create(names.getTaskName("link"), LinkBundle.class);
+                    link.source(compile.getObjectFileDirectory().getAsFileTree().matching(new PatternSet().include("**/*.obj", "**/*.o")));
+                    link.lib(binary.getLinkLibraries());
+                    link.setLinkerArgs(Collections.<String>emptyList());
+                    final PlatformToolProvider toolProvider = ((NativeToolChainInternal) toolChain).select(currentPlatform);
+                    Provider<RegularFile> exeLocation = buildDirectory.file(providers.provider(new Callable<String>() {
+                        @Override
+                        public String call() {
+                            return toolProvider.getExecutableName("exe/" + names.getDirName() + binary.getModule().get());
```

Comment:
**Open question:**
- Should we also add modeling to `PlatformToolProvider` for bundle binary name?
- Should the bundle binary placed in `exe` folder?
- Should we compose a bundle in 2 stage: 1) build the main bundle binary and 2) assemble the bundle by copying the bundle to the right location? Or should we output the binary to the right location right away?
  - Outputing the binary to the right location would at the very least cause 2 tasks outputing in the same folder. We could rename the `LinkBundle` task to `CreateBundle` and compose the bundle from that task directly.

Reply:
It feels like something that should be done as 2 steps.